src/main.rs

```diff
@@ -94,6 +138,14 @@ impl manager::NotificationWatcher for Program {
     }
 }
 
+fn print_controller(controller: &controller::Controller) {
```

Comment:
Yeah - that's a possibility.

I made values print the home ID and id as hex. We should probably do that here for consistency.